File: Embodied-AI/ROS/catkin_ws/src/ssr_pkg/src/chao_node.cpp
#include<ros/ros.h>          // ROS的核心功能头文件
#include<std_msgs/String.h>  // 字符串消息类型的头文件
#include<qq_msgs/Carry.h>
int main(int argc, char  *argv[])
{   

    ros::init(argc,argv,"chao_node");// 初始化ROS节点，节点名为"chao_node"
    ros::NodeHandle nh;  // 创建节点句柄，用于管理节点资源
    ros::Publisher pub = nh.advertise<qq_msgs::Carry>("chao",10);

    ros::Rate loop_rate(10);// 设置循环频率为10Hz
    while(ros::ok())//使用ture时候运行节点不能响应外界信号
    {
        printf("keep\n");
        qq_msgs::Carry msg; // 创建字符串消息对象
        msg.grade="man";
        msg.star=100;
        msg.data = "do this all day man!!!"; // 设置消息内容
        pub.publish(msg);// 发布消息
        loop_rate.sleep(); // 控制循环频率
    }
    printf("hello world");
    return 0;
}

File: Embodied-AI/ROS/catkin_ws/src/ssr_pkg/scripts/chao_node.py
#! /usr/bin/env python3
#coding=utf-8

import rospy
from std_msgs.msg import String
from qq_msgs.msg import Carry
if __name__=="__main__":
    rospy.init_node("chao_node")
    pub=rospy.Publisher("chao",Carry,queue_size=10)
    rate=rospy.Rate(10)
    
    while not rospy.is_shutdown():
        msg=Carry()
        msg.grade="man"
        msg.star=100
        msg.data="do this all day man!!!"
        pub.publish(msg)
        rospy.loginfo("man!!!")
        rate.sleep()
